synadm/cli/__init__.py

```diff
@@ -132,13 +136,16 @@ def output(self, data):
     default="~/.config/synadm.yaml",
     help="configuration file path", show_default=True)
 @click.pass_context
-def root(ctx, verbose, output, config_file):
+def root(ctx, verbose, batch, output, config_file):
     """ Synapse Administration toolkit
     """
-    ctx.obj = APIHelper(config_file, verbose, output)
+    ctx.obj = APIHelper(config_file, verbose, batch, output)
     if ctx.invoked_subcommand != "config" and not ctx.obj.load():
```

Comment:
Seems like ctx.obj.load() is not executed here and thus calling `synadm config` does not read from config file.
Something like `helper_load_success = ctx.obj.load()` and then asking for this variable in the condition fixes it.

Reply:
Also after writing new config to file it needs to be read again here otherwise if config was called automatically because a user issued any command with an incomplete /broken config file, after going through the configurator it would fail. Putting this after the write fixes it: `helper.config.load()`